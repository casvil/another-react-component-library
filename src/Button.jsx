import React from "react";

const Button = ({ disabled, kind, onClick, size, style, value }) => (
  <input
    className={`bg-${kind}-500 text-${kind} px-3 py-2`}
    type="button"
    value={value}
    onClick={onClick}
    disabled={disabled}
  />
);

export default Button;
