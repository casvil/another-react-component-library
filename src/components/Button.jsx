import React from "react";
import PropTypes from "prop-types";

const Button = ({ disabled, kind, onClick, size, style, value }) => (
  <input
    className={`
      active:bg-${kind}-400 
      ${disabled ? `` : `hover:bg-${kind}-700`} 
      bg-${kind}-500 
      focus:outline-none 
      px-3 
      py-2 
      rounded
      text-${kind} 
      ${disabled ? `bg-opacity-75` : `bg-opacity-100`}
      ${disabled ? `cursor-not-allowed` : `cursor-pointer`}
    `}
    type="button"
    value={value}
    onClick={onClick}
    disabled={disabled}
  />
);

Button.propTypes = {
  disabled: PropTypes.bool,
  kind: PropTypes.oneOf(["default", "primary", "success", "warning", "danger"]),
};

Button.defaultProps = {
  disabled: false,
  kind: "default",
};

export default Button;
