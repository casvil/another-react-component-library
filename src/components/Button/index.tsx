import React from "react";
import clsx from "clsx";

export type ButtonVariant = "primary" | "secondary" | "outline" | "danger";
export type ButtonSize = "sm" | "md" | "lg";

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  size?: ButtonSize;
  loading?: boolean;
}

const baseStyles = "rounded-xl font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none";
const variantStyles: Record<ButtonVariant, string> = {
  primary: "bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500",
  secondary: "bg-gray-100 text-gray-900 hover:bg-gray-200 focus:ring-gray-400",
  outline: "border border-gray-300 text-gray-700 bg-white hover:bg-gray-50 focus:ring-gray-400",
  danger: "bg-red-600 text-white hover:bg-red-700 focus:ring-red-500",
};
const sizeStyles: Record<ButtonSize, string> = {
  sm: "px-3 py-1.5 text-sm",
  md: "px-4 py-2 text-base",
  lg: "px-5 py-3 text-lg",
};

export const Button: React.FC<ButtonProps> = ({
  children,
  variant = "primary",
  size = "md",
  loading = false,
  className,
  disabled,
  ...props
}) => {
  const classes = clsx(
    baseStyles,
    variantStyles[variant],
    sizeStyles[size],
    className
  );

  return (
    <button
      className={classes}
      disabled={disabled || loading}
      {...props}
    >
      {loading ? <span className="animate-spin h-4 w-4 border-2 border-t-transparent border-white rounded-full inline-block mr-2" /> : null}
      {children}
    </button>
  );
};
