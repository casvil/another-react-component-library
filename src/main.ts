export { Button } from './components/Button'
// export { MyCounter } from './components/MyCounter'
// export { MyTitle } from './components/MyTitle'
