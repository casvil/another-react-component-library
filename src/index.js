import Button from "./components/atoms/Button";
import Select from "./components/atoms/Select";

export { Button, Select };
