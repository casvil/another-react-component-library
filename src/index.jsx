import React from "react";
import ReactDOM from "react-dom";
import "../public/index.css";
import "tailwindcss/dist/tailwind.css";

import Button from "./Button.jsx";

ReactDOM.render(
  <React.Fragment>
    <React.StrictMode>
      <div className="p-5 bg-purple-500 text-white">
        <h1 className="mt-4 mb-4 text-5xl">Another React Component Library</h1>
      </div>
      <div className="m-5">
        <h1 className="mb-2 text-2xl">Buttons</h1>
        <h2 className="mb-1 mt-3 text-lg">Contained Buttons</h2>
        <div className="inline-block mr-3">
          <Button kind="default" value="Default" />
        </div>
        <div className="inline-block mr-3">
          <Button kind="primary" value="Primary" />
        </div>
        <div className="inline-block mr-3">
          <Button kind="success" value="Success" />
        </div>
        <div className="inline-block mr-3">
          <Button kind="warning" value="Warning" />
        </div>
        <div className="inline-block mr-3">
          <Button kind="danger" value="Danger" />
        </div>
        <h2 className="mb-1 mt-3 text-lg">Disabled</h2>
        <div className="inline-block mr-3">
          <Button disabled={true} kind="default" value="Default" />
        </div>
        <div className="inline-block mr-3">
          <Button disabled={true} kind="primary" value="Primary" />
        </div>
        <div className="inline-block mr-3">
          <Button disabled={true} kind="success" value="Success" />
        </div>
        <div className="inline-block mr-3">
          <Button disabled={true} kind="warning" value="Warning" />
        </div>
        <div className="inline-block mr-3">
          <Button disabled={true} n kind="danger" value="Danger" />
        </div>
      </div>
    </React.StrictMode>
  </React.Fragment>,
  document.getElementById("root")
);

// Hot Module Replacement (HMR) Learn more: https://www.snowpack.dev/concepts/hot-module-replacement
if (import.meta.hot) {
  import.meta.hot.accept();
}
