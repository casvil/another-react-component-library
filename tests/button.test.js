import React from "react";
import renderer from "react-test-renderer";

import { Button } from "../src/components";

test("Renders a default button", () => {
  const component = renderer.create(<Button kind="default" value="Default" />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
