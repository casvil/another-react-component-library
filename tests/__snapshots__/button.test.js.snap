// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a default button 1`] = `
<input
  className="
      active:bg-default-400 
      hover:bg-default-700 
      bg-default-500 
      focus:outline-none 
      px-3 
      py-2 
      rounded
      text-default 
      bg-opacity-100
      cursor-pointer
    "
  disabled={false}
  type="button"
  value="Default"
/>
`;
